app/commands/exercise/representation/send_new_feedback_notifications.rb

```diff
@@ -0,0 +1,36 @@
+class Exercise::Representation::SendNewFeedbackNotifications
+  include Mandate
+
+  queue_as :notifications
+
+  initialize_with :representation
+
+  def call = iterations.each { |iteration| send_notification(iteration) }
+
+  private
+  def iterations
+    if representation.has_essential_feedback?
+      latest_active_iterations
+    elsif representation.has_actionable_feedback?
+      latest_recent_active_iterations
+    else
+      raise "Incorrect feedback type"
```

Comment:
I'd have a small preference for this being a noop and removing the guard below. Then we're encapsulating the knowledge into one place, whereas currently we have two different places we have to remember to keep in sync.

Reply:
Excellent suggestion